### ISSUE TITLE
External Signaling Server - Error: Can not connect to server - No Logs

### ISSUE THREAD

**Issue by Giom-fm** [nextcloud/spreed#5687]
## Steps to reproduce
1. Go to the Task settings page
2. Paste your domain name and your secret of your external signaling server
3. Wait until the connection has been checked

### Expected behaviour
The connection should be established or at least a reason should be given in the logs why this is not the case.

### Actual behaviour
The connection could not be established. The only error is: "Error: Can not connect to server"
![image](https://user-images.githubusercontent.com/35076723/120081767-8950ea00-c0bf-11eb-8349-c52cda7b78da.png)

What I find strange is that although the UI says that the server connection is being tested, I can't find anything in the reverse-proxy logs of the external signaling server. So I assume that no attempt is made to contact the server.

If I test the server via following command I can established an connection
![image](https://user-images.githubusercontent.com/35076723/120081899-30ce1c80-c0c0-11eb-8dbc-8668b4abcb72.png)

Additionally I see entries in the log of the reverse-proxy
![image](https://user-images.githubusercontent.com/35076723/120082095-57d91e00-c0c1-11eb-8572-9ae37a4f7d09.png)

It could be that I have configured something incorrectly, however, although an error is clearly displayed in the UI, I can't find an entry for it in the nextcloud.log, even though I have set the verbosity to debug. Thus I can neither triangulate nor eliminate the error.

## Talk app

**Talk app version:** 10.0.7

**Custom Signaling server configured:** [yes](https://github.com/strukturag/nextcloud-spreed-signaling) as docker container

**Custom TURN server configured:** no 

**Custom STUN server configured:** no 

## Server configuration

**Operating system**: debian 9 - All Services running in docker containers

**Web server:** nginx/1.19.10 (fpm-fcgi)

**Database:** mysql 10.5.10

**PHP version:** 7.4.19

**Nextcloud Version:** 20.0.10 - 20.0.10.1

**List of activated apps:**

<details>

```
Enabled:
  - accessibility: 1.6.0
  - activity: 2.13.4
  - cloud_federation_api: 1.3.0
  - comments: 1.10.0
  - contactsinteraction: 1.1.0
  - dashboard: 7.0.0
  - dav: 1.16.2
  - federatedfilesharing: 1.10.2
  - federation: 1.10.1
  - files: 1.15.0
  - files_pdfviewer: 2.0.1
  - files_rightclick: 0.17.0
  - files_sharing: 1.12.2
  - files_trashbin: 1.10.1
  - files_versions: 1.13.0
  - files_videoplayer: 1.9.0
  - firstrunwizard: 2.9.0
  - issuetemplate: 0.7.0
  - logreader: 2.5.0
  - lookup_server_connector: 1.8.0
  - nextcloud_announcements: 1.9.0
  - notifications: 2.8.0
  - oauth2: 1.8.0
  - password_policy: 1.10.1
  - photos: 1.2.3
  - privacy: 1.4.0
  - provisioning_api: 1.10.0
  - recommendations: 0.8.0
  - serverinfo: 1.10.0
  - settings: 1.2.0
  - sharebymail: 1.10.0
  - spreed: 10.0.7
  - support: 1.3.0
  - survey_client: 1.8.0
  - systemtags: 1.10.0
  - text: 3.1.0
  - theming: 1.11.0
  - twofactor_backupcodes: 1.9.0
  - updatenotification: 1.10.0
  - user_status: 1.0.1
  - viewer: 1.4.0
  - weather_status: 1.0.0
  - workflowengine: 2.2.0
Disabled:
  - admin_audit
  - encryption
  - files_external
  - user_ldap
```
</details>

**Nextcloud configuration:**

<details>

```
{
    "system": {
        "memcache.local": "\\OC\\Memcache\\APCu",
        "apps_paths": [
            {
                "path": "\/var\/www\/html\/apps",
                "url": "\/apps",
                "writable": false
            },
            {
                "path": "\/var\/www\/html\/custom_apps",
                "url": "\/custom_apps",
                "writable": true
            }
        ],
        "memcache.distributed": "\\OC\\Memcache\\Redis",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "password": "***REMOVED SENSITIVE VALUE***",
            "port": 6379
        },
        "overwriteprotocol": "https",
        "trusted_proxies": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "localhost",
            ""***REMOVED SENSITIVE VALUE***"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "mysql",
        "version": "20.0.10.1",
        "overwrite.cli.url": "https:\/\/localhost",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "log_type": "file",
        "logfile": "nextcloud.log",
        "loglevel": 0
    }
}
```
</details>

### Server log (data/nextcloud.log)
I wish I had some...

**Comment by nickvergessen** [comment]
Did you do your test from the nextcloud server to the signaling server?
Also as you mention reverse proxy make sure to have all ports forwarded.

The nextcloud.log should give you info, but you misconfigured it in your config.php (seems to contain no path)

Also you removed all domain information so we can not even see what a potential problem could be. 🤷‍♂️

**Comment by Giom-fm** [comment]
Thanks for you quick reply.

For testing purpose everything is on the same server.
All my tests (screenshots) are from that server. I just added another subdomain for the signaling server.
As mentioned [here](https://codeberg.org/wh0ami/signaling-dockerized#why-shouldnt-i-use-the-docker-compose-stack-from-the-original-nextcloud-spreed-signaling-maintainer) and according to my understanding I do not have to forward any ports.
Please correct me if Im wrong.

> The nextcloud.log should give you info, but you misconfigured it in your config.php (seems to contain no path)

The file is created and also filled with some inputs. But they do not seem to be connected to the actual problem.
I have attached it. [nextcloud.log](https://github.com/nextcloud/spreed/files/6565290/nextcloud.log)

**Comment by nickvergessen** [comment]
> Did you do your test from the nextcloud server to the signaling server?

Okay, let me rephrase:
Did you do your test from the nextcloud container to the signaling container?

Quoting https://codeberg.org/wh0ami/signaling-dockerized#why-shouldnt-i-use-the-docker-compose-stack-from-the-original-nextcloud-spreed-signaling-maintainer
> All containers are in the same network and can communicate which each other, which is also not required

Well it is required that Nextcloud and the signaling can communicate with each other.

I would recomment to try it with the provided image, if that works your thing is doing something wrong.

> Error: Cannot connect to server

Means the Nextcloud server can not contact the the HPB. This is not a bug in Talk nor the HPB, so please search for further help at your docker provider or the forum.
